src/Consolonia.Core/Drawing/PixelBufferImplementation/Pixel.cs

```diff
@@ -102,56 +102,43 @@ public bool Equals(Pixel other)
             PixelForeground newForeground;
-            PixelBackground newBackground;
 
-            if (pixelAbove.IsCaret) return new Pixel(Foreground, Background, true);
+            var newBackground = new PixelBackground(MergeColors(Background.Color, pixelAbove.Background.Color));
 
-            switch (pixelAbove.Background.Mode)
+            bool newIsCaret;
+
+            if (pixelAbove.Background.Color.A == 0x0 /*todo: can be approximate, extract to extension method*/)
+            {
+                newForeground = pixelAbove.Foreground.NothingToDraw()
+                    ? Foreground
+                    : Foreground.Blend(pixelAbove.Foreground);
+                newIsCaret = pixelAbove.IsCaret | IsCaret;
+            }
+            else
             {
-                case PixelBackgroundMode.Colored:
-                    // merge pixelAbove into this pixel using alpha channel.
-                    Color mergedColors = MergeColors(Background.Color, pixelAbove.Background.Color);
+                if (!pixelAbove.Foreground.NothingToDraw())
+                {
                     newForeground = pixelAbove.Foreground;
-                    newBackground = new PixelBackground(mergedColors);
-                    return new Pixel(newForeground, newBackground, pixelAbove.IsCaret);
-
-                case PixelBackgroundMode.Transparent:
-                    // if the foreground is transparent, ignore pixelAbove foreground.
-                    newForeground = pixelAbove.Foreground.Color != Colors.Transparent
-                        ? Foreground.Blend(pixelAbove.Foreground)
-                        : Foreground;
-
-                    // background is transparent, ignore pixelAbove background.
-                    newBackground = Background;
-                    break;
-                case PixelBackgroundMode.Shaded:
-                    // shade the current pixel
-                    (newForeground, newBackground) = Shade();
-
-                    // blend the pixelAbove foreground into the shaded pixel
-                    newForeground = newForeground.Blend(pixelAbove.Foreground);
-
-                    // resulting in new pixel with shaded background and blended foreground
-                    return new Pixel(newForeground, newBackground);
-
-                default: throw new ArgumentOutOfRangeException(nameof(pixelAbove));
+                }
+                else
+                {
+                    newForeground = new PixelForeground(Foreground.Symbol,
+                        MergeColors(Foreground.Color, pixelAbove.Background.Color),
+                        Foreground.Weight,
+                        Foreground.Style,
+                        Foreground.TextDecoration);
+
+                    if (pixelAbove.Background.Color.A == 0xFF)
+                        // non-transparent layer above
+                        newForeground = new PixelForeground();
+                }
+
+                newIsCaret = pixelAbove.IsCaret;
```

Comment:
not this delta but I think we can greatly simplify the logic by pulling IsCaret off of Pixel entirely and move to property of PixelBuffer
```PixelBuffer.Caret = X/Y coordate```
We are using caret for focus, and Avalonia never has more than a single focus. And Console Apps don't (as far as I know) have the ability to have multiple carets. 

Why not save ourselves the complexity (and even memory) ? 

But we should do this on a different PR

Reply:
I'm not yet convinced it would be better. Caret covered by a background or foreground is supposed to dissapear similar to a foreground of a foreground below. How would this look like? lets try

Reply:
I see. Sigh